src/cli/cli_srp_client.cpp

```diff
@@ -404,6 +405,35 @@ otError SrpClient::ProcessServiceAdd(Arg aArgs[])
     string = otSrpClientBuffersGetServiceEntryServiceNameString(entry, &size);
     SuccessOrExit(error = CopyString(string, size, aArgs[2].GetCString()));
 
+    // Service subtypes are added as part of service name as a comma separated list
+    // e.g., "_service._udp,_sub1,_sub2"
+
+    label = strchr(string, ',');
+
+    if (label != nullptr)
+    {
+        uint16_t     arrayLength;
+        const char **subTypeLabels = otSrpClientBuffersGetSubTypeLabelsArray(entry, &arrayLength);
+
+        // Leave the last array element as `nullptr` to indicate end of array.
+        for (uint16_t index = 0; index + 1 < arrayLength; index++)
+        {
+            *label++             = '\0';
+            subTypeLabels[index] = label;
+
+            label = strchr(label, ',');
+
+            if (label == nullptr)
+            {
+                break;
```

Comment:
Do we need to explicitly set `subTypeLabels[index + 1]` to `nullptr`?

Reply:
We can consider adding it but it is not technically necessary since `otSrpClientBuffersAllocateService()` will ensure all buffers and strings are cleared on the newly allocated entry. I will add a comment in `otSrpClientBuffersAllocateService()` to mention this.